lib/schema.js

```diff
@@ -213,10 +213,24 @@ Schema.prototype.refKeys = function(mdl) {
   for (var i = 0; i < refIndices.length; ++i) {
     var refIndex = refIndices[i];
     var refValues = [];
+    // This value will remain false unless a relevant value is found
+    var defined = false;
     for (var j = 0; j < refIndex.fields.length; ++j) {
-      refValues.push(this.fieldVal(mdl, refIndex.fields[j]));
+      var value = this.fieldVal(mdl, refIndex.fields[j]);
+      refValues.push(value);
+      //Prevents refdoc keys from having "undefined"
+      if(value){
```

Comment:
Also, you can probably optimize this section by performing the inverse of what you're doing (instead of triggering success on non-undefined, you should trigger failure on an undefined field).
```javascript
    var refIndex = refIndices[i];
    var refValues = [];
    for (var j = 0; j < refIndex.fields.length; ++j) {
      var value = this.fieldVal(mdl, refIndex.fields[j]);
      if (value === undefined) {
        // If we encounter an undefined value, we should skip this refkey.
        refValues = [];
        break;
      }
      refValues.push(value);
    }
    if (refValues.length > 0) {
      refKeys.push(this.refKey(refIndex.fields, refValues));
    }
```